qemu/tests/physical_resources_check.py

```diff
@@ -207,12 +208,8 @@ def verify_machine_type():
     if params.get("catch_serial_cmd") is not None:
         length = int(params.get("length", "20"))
         id_leng = random.randint(0, length)
-        drive_serial = ""
-        convert_str = "!\"#$%&\'()*+./:;<=>?@[\\]^`{|}~"
-        drive_serial = utils_misc.generate_random_string(id_leng,
-                                                         ignore_str=",",
-                                                         convert_str=convert_str)
-
+        ignore_str = string.punctuation.replace("-", "").replace("_", "")
```

Comment:
`import string

result = string.punctuation
print(result)
!"#$%&'()*+,-./:;<=>?@[\]^_`{|}~`

Reply:
OK, I see. Agree with you, select drive_serial characters from ascii_letters + digits + "-"  + "_".